packages/kbn-shared-ux-components/src/empty_state/kibana_no_data_page.tsx

```diff
@@ -17,19 +18,27 @@ export interface Props {
 
 export const KibanaNoDataPage = ({ onDataViewCreated, noDataConfig }: Props) => {
   const { hasESData, hasUserDataView } = useData();
+  const [isLoading, setIsLoading] = useState(true);
   const [dataExists, setDataExists] = useState(false);
   const [hasUserDataViews, setHasUserDataViews] = useState(false);
 
   useEffect(() => {
     const checkData = async () => {
       setDataExists(await hasESData());
       setHasUserDataViews(await hasUserDataView());
+      setIsLoading(false);
     };
     // TODO: add error handling
     // https://github.com/elastic/kibana/issues/130913
-    checkData().catch(() => {});
+    checkData().catch(() => {
+      setIsLoading(false);
+    });
   }, [hasESData, hasUserDataView]);
 
+  if (isLoading) {
+    return <EuiLoadingElastic css={{ alignSelf: 'center', justifySelf: 'center' }} size="xxl" />;
```

Comment:
Looks like just margin will actually do the trick here

```suggestion
    return < EuiLoadingElastic css={{ margin: 'auto' }} size="xxl" />;
```